### PR TITLE
[DEV-519] kyc code

### PR DESCRIPTION
**Pull request by Krysh90**
which is up for discussion is how we are showing blanked information to the user. I have made a proposal, but added a comment, which should be removed after a decision was made

### CODE REVIEW

**Comment by Krysh90** [comment]
> which is up for discussion is how we are showing blanked information to the user. I have made a proposal, but added a comment, which should be removed after a decision was made

decided on a functionality.
We are showing a disabled input field if blanked information is available. There is an edit button right besides those disabled fields, if a user decides that he wants to edit those information. Then default fields are being displayed and the user is able to edit the information

File: app/services/ApiService.ts [hunk 1]
```diff
@@ -89,24 +89,26 @@ export const deleteApiKey = (): Promise<void> => {
 };
 
 // --- KYC --- //
-export const putKycData = (data: KycData): Promise<void> => {
-  return fetchFrom(`${KycUrl}/data`, "POST", toKycDataDto(data));
+
+// TODO: check these services if called correctly with code
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply app/services/ApiService.ts:93]: actually was done, but not removed.

File: app/screens/KycScreen.tsx [hunk 2]
```diff
@@ -37,42 +42,62 @@ const KycScreen = ({ settings }: { settings?: AppSettings }) => {
...
+    nav.navigate(Routes.Kyc, { code: undefined, autostart: undefined, phone: undefined, mail: undefined });
 
     // get KYC info
     getKyc(params?.code)
       .then((result) => {
-        updateState(result);
-        if (params?.autostart) onContinue(result);
+        updateState(result, params);
+        if (params?.autostart && result.kycStatus !== KycStatus.NA) onContinue(result);
+        else if (params?.autostart && shouldStart(result)) requestStart(params?.code);
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply app/screens/KycScreen.tsx:71]: yes agree it is kinda complicated with this logic being split into two different functions

File: app/screens/KycScreen.tsx [hunk 3]
```diff
@@ -37,42 +42,62 @@ const KycScreen = ({ settings }: { settings?: AppSettings }) => {
...
+  const [hasUploadedFile, setHasUploadFile] = useState(false);
+  const [isAutostart, setIsAutostart] = useState(false);
 
   useEffect(() => {
     // get params
     const params = route.params as any;
     if (!params?.code) return onLoadFailed();
 
     setCode(params.code);
+    setIsAutostart(params.autostart);
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply app/screens/KycScreen.tsx:61]: Agree, makes it a whole lot easier. I thought of autostart, to reduce the amount of clicks to be made. I am perfectly fine to just use it once.

File: app/screens/KycScreen.tsx [hunk 4]
```diff
@@ -117,49 +181,95 @@ const KycScreen = ({ settings }: { settings?: AppSettings }) => {
...
-              </CompactRow>
-            </DataTable>
-            <SpacerV />
-            {/* TODO: integrate initial limit increase */}
-            {((kycInProgress(kycResult.kycStatus) && kycResult.kycState !== KycState.REVIEW) ||
-              kycCompleted(kycResult.kycStatus)) && (
+          <>
+            <Portal>
+              <Dialog
+                visible={showsUploadDialog && kycData?.accountType === AccountType.BUSINESS && !hasUploadedFile}
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply app/screens/KycScreen.tsx:220]: ah yes, should have been removed. I have added `showsUploadDialog` and forgot to remove that `&& BUSINESS` check. thanks

**Review by davidleomay — APPROVED** [review]

File: app/services/ApiService.ts [hunk 5]
```diff
@@ -89,24 +89,24 @@ export const deleteApiKey = (): Promise<void> => {
 };
 
 // --- KYC --- //
-export const putKycData = (data: KycData): Promise<void> => {
-  return fetchFrom(`${KycUrl}/data`, "POST", toKycDataDto(data));
+export const putKycData = (data: KycData, code?: string): Promise<KycInfo> => {
```
Comment (ThaBrad) [inline app/services/ApiService.ts:92]: Why is the code an optional param? (Just realising it now)
If it isn't initialised the fetch is …/kyc/**undefined**/data

This **ALWAYS** fails if code is not defined!

-->
```
  "message": "Error: Conversion failed when converting from a character string to uniqueidentifier.",
  "statusCode": 500,
```

**Pull request merged**